rs/cli/src/commands/mod.rs

```diff
@@ -48,24 +50,71 @@ pub mod upgrade;
 mod version;
 mod vote;
 
-#[derive(Parser, Debug)]
-#[clap(version = env!("CARGO_PKG_VERSION"), about, author)]
-pub struct Args {
-    /// Pin for the HSM key used for submitting proposals
-    #[clap(long, global = true, hide_env_values = true, env = "HSM_PIN")]
-    pub hsm_pin: Option<String>,
-
+/// HSM authentication parameters
+#[derive(ClapArgs, Debug, Clone)]
+pub(crate) struct HsmParams {
     /// Slot that HSM key uses, can be read with pkcs11-tool
-    #[clap(long, value_parser=maybe_hex::<u64>, global = true, env = "HSM_SLOT")]
-    pub hsm_slot: Option<u64>,
+    #[clap(required = false,
+        conflicts_with = "private_key_pem",
+        long, value_parser=maybe_hex::<u64>, global = true, env = "HSM_SLOT")]
+    pub(crate) hsm_slot: Option<u64>,
 
     /// HSM Key ID, can be read with pkcs11-tool
-    #[clap(long, global = true, env = "HSM_KEY_ID")]
-    pub hsm_key_id: Option<String>,
+    #[clap(required = false, conflicts_with = "private_key_pem", long, value_parser=maybe_hex::<u8>, global = true, env = "HSM_KEY_ID")]
```

Comment:
What's the use case for the `conflicts_with`?

Reply:
It rejects the use of the flag if the other flag is present.

Reply:
Technically we could use groups for this, but groups aren't working at all here, for some reason.